src/simsopt/mhd/normal_field.py

```diff
@@ -0,0 +1,306 @@
+from .._core.util import nested_lists_to_array
+from .._core.optimizable import DOFs, Optimizable
+import logging
+
+import numpy as np
+from scipy.io import netcdf
+from scipy.interpolate import interp1d
+import f90nml
+from monty.json import MSONable
+
+logger = logging.getLogger(__name__)
+
+try:
+    import py_spec
+except ImportError as e:
+    py_spec = None
+    logger.debug(str(e))
+
+
+__all__ = ['SurfaceRZFourier', 'SurfaceRZPseudospectral']
+
+
+class NormalField(Optimizable):
+    r"""
+    ``NormalField`` represents the normal field on a boundary, for example the
+    computational boundary of SPEC free-boundary.
+
+    Args:
+        nfp: The number of field period
+        stellsym: Whether (=True) or not (=False) stellarator symmetry is enforced.
+        mpol: Poloidal Fourier resolution
+        ntor: Toroidal Fourier resolution
+        vns: Odd fourier modes of :math: `\mathbf{B}\cdot\mathbf{\hat{n}}`. 2D array of size
+          (mpol+1)x(2ntor+1). Set to None to fill with zeros
+        vnc: Even fourier modes of :math: `\mathbf{B}\cdot\mathbf{\hat{n}}`. 2D array of size
+          (mpol+1)x(2ntor+1). Ignored if stellsym if True. Set to None to fill with zeros
+    """
+
+    def __init__(self, nfp=1, stellsym=True, mpol=1, ntor=0,
+                 vns=None, vnc=None):
+
+        self.nfp = nfp
+        self.stellsym = stellsym
+        self.mpol = mpol
+        self.ntor = ntor
+
+        if vns is None:
+            self.vns = np.zeros((self.mpol + 1, 2 * self.ntor + 1))
+        else:
+            self.vns = vns[0:self.mpol + 1][-self.ntor:self.ntor]
```

Comment:
That was, in my opinion, more practical (the index is equal to the mode number) but I understand it can generate some errors. I modified it as suggested.